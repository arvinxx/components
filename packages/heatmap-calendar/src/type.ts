/**
 * 热力图数据结构
 */
export interface ItemData {
  /**
   * 日期
   */
  date: string;
  /**
   * 天数
   */
  day: number;
  /**
   * 月份
   */
  month: number;
  /**
   * 热点数
   */
  total: number;
  /**
   * 周
   */
  week: number;
  /**
   * 年
   */
  year: number;
}

export type HeatmapCalendarData = ItemData[];
