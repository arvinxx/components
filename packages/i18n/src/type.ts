import type { ComponentProps, FC } from 'react';
import type { IntlProvider } from 'react-intl';

export type LocalesType = 'en-US' | 'zh-CN';

export type LocaleKey = string;

export type IntlProviderProps<T = Record<string, string>> = FC<
  Omit<ComponentProps<typeof IntlProvider>, 'messages'> & {
    messages: T;
  }
>;
