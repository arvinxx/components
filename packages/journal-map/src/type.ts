/**
 * 阶段
 */
export interface Step<T = string> {
  id: T;
  name: string;
  /**
   * 该步骤的颜色
   */
  color?: string;
}

/**
 * 用户旅程地图数据
 */
export interface JournalMapData<T extends string = string> {
  /**
   * 大的步骤
   */
  steps: Step<T>[];
  /**
   * 用户行为
   */
  actions: Record<T, UserAction[]>;
}

/**
 * 用户行为
 */
export interface UserAction {
  /**
   * 行为名称
   */
  title: string;
  /**
   * 说明
   */
  description?: string;

  /**
   * 有关的想法
   */
  thoughts?: string[];
  /**
   * 情绪得分
   */
  emotion: number;
  /**
   * 痛点
   */
  painSpot?: string[];
}
