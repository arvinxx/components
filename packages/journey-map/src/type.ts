/**
 * 阶段
 */
export interface Step<T = string> {
  /**
   * 阶段id
   * @TJS-type string
   */
  id: T;
  /**
   * 阶段名称
   */
  name: string;
  /**
   * 该阶段的颜色
   */
  color?: string;
}

/**
 * 用户旅程地图数据
 */
export interface JourneyMapData<T extends string = string> {
  /**
   * 大的步骤
   */
  steps: Step<T>[];
  /**
   * 用户行为
   */
  actions: Record<T, UserAction[]>;
}

/**
 * 用户行为
 */
export interface UserAction {
  /**
   * 行为名称
   */
  title: string;
  /**
   * 说明
   */
  description?: string;

  /**
   * 有关的想法
   */
  thoughts?: string[];
  /**
   * 情绪得分
   */
  emotion: number;
  /**
   * 痛点
   */
  painSpot?: string[];
}

/**
 * 解析后的 YML 数据
 */
export interface JourneyMapYML<T extends string = string> {
  /**
   * 步骤
   */
  steps: Step<T>[];
  /**
   * 用户行为
   */
  actions: Record<T, Record<T, number>[]>;
}
