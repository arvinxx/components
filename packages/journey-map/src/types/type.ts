/**
 * 阶段
 */
export interface Stage<T = string> {
  /**
   * 阶段id
   * @TJS-type string
   */
  id: T;
  /**
   * 阶段名称
   */
  name: string;
  /**
   * 该阶段的颜色
   */
  color?: string;
}

/**
 * 用户旅程地图数据
 */
export interface JourneyMapData<T extends string = string> {
  /**
   * 大的步骤
   */
  stages: Stage<T>[];
  /**
   * 用户行为
   */
  actions: Record<T, UserAction[]>;
}

/**
 * 用户行为类型
 */
export interface UserAction {
  /**
   * 行为名称
   */
  name: string;
  /**
   * 情绪得分
   * @minimum -2
   * @maximum 2
   * @TJS-type integer
   */
  emotion?: number;

  /**
   * 说明
   */
  description?: string;

  /**
   * 想法
   */
  thoughts?: string[];

  /**
   * 痛点
   */
  painSpot?: string[];
}
