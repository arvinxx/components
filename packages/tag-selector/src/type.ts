export interface Value {
  key: string;
  text: string;
  refText?: string;
  value?: number;
  editable?: boolean;
}

export interface Item {
  key: string;
  title: string;
  value: Value[];
  inputVisible?: boolean;
}
