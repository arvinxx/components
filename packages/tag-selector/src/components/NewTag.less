@import '../theme/token';

.input {
  align-self: center;
  width: 80px;
  background-color: @base-white;
  border-radius: 4px;
  box-shadow: none;
}

.plus {
  align-self: center;
  width: 24px;
  margin-right: 0;
  padding-left: 5px;
  color: @transBlack-4;
  background: @base-white;
  border-color: @transBlack-6;
  &:hover {
    color: @transBlack-2;
  }
  &:active {
    color: @base-black;
  }
}
