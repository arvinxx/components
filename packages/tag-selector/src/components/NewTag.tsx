import React, { PureComponent } from 'react';
import { Tag, Input } from 'antd';
import { PlusOutlined } from '@ant-design/icons';

import styles from './NewTag.less';

export interface INewTagProps {
  labelKey: string;
  inputVisible: boolean;
}
export default class NewTag extends PureComponent<INewTagProps> {
  static defaultProps: INewTagProps = {
    labelKey: '',
    inputVisible: false,
  };
  state = { text: '' };

  newTagOnInput = (e) => {
    this.setState({ text: e.target.value });
  };

  handleKeyDown = (e, key) => {
    console.log(key);
    // 按 ESC 撤销修改并退出
    if (e.key === 'Escape') {
      // this.props.dispatch({
      //   type: 'label/hideTagInput',
      //   payload: key,
      // });
      this.setState({ text: '' });
    }

    // 修改完毕后按 Tab 保存修改,并继续创建新标签
    // 修改完毕后按 Enter 保存修改,并结束创建标签
    if (e.key === 'Tab' || e.key === 'Enter') {
      e.preventDefault();
      const { text } = this.state;
      if (text !== '') {
        // this.props.dispatch({
        //   type: 'label/addTag',
        //   payload: { key, text },
        // });
      }
      if (e.key === 'Enter') {
        // this.props.dispatch({
        //   type: 'label/hideTagInput',
        //   payload: key,
        // });
      }
      this.setState({ text: '' });
    }
  };

  showTagInput = (key) => {
    console.log(key);
    // this.props.dispatch({
    //   type: 'label/showTagInput',
    //   payload: key,
    // });
  };
  newTagOnBlur = (key) => {
    const { text } = this.state;
    console.log(key);
    if (text !== '') {
      // this.props.dispatch({
      //   type: 'label/addTag',
      //   payload: { key, text },
      // });
    }
    // this.props.dispatch({
    //   type: 'label/hideTagInput',
    //   payload: key,
    // });
    this.setState({ text: '' });
  };

  render() {
    const { labelKey: key, inputVisible } = this.props;
    const { text } = this.state;

    return inputVisible ? (
      <Input
        key={`${key}-add`}
        type="text"
        size="small"
        autoFocus={inputVisible}
        className={styles.input}
        value={text}
        onKeyDown={(e) => this.handleKeyDown(e, key)}
        onChange={this.newTagOnInput}
        onBlur={() => this.newTagOnBlur(key)}
      />
    ) : (
      <Tag
        key={`${key}-plus`}
        onClick={() => this.showTagInput(key)}
        className={styles.plus}
      >
        <PlusOutlined />
      </Tag>
    );
  }
}
