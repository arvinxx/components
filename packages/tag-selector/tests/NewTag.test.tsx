import React from 'react';

import { render } from '@testing-library/react';
import type { INewTagProps } from '../src/components/NewTag';
import NewTag from '../src/components/NewTag';

const props: INewTagProps = {
  labelKey: 'rercxc',
  inputVisible: false,
};

describe('Tag', () => {
  it('render', () => {
    const { container } = render(<NewTag {...props} />);

    expect(container).toMatchSnapshot();
  });
  // describe('交互行为', () => {
  //   describe('点击加号按钮添加新标签', () => {
  //     it('showValueInput should run when change', () => {
  //       wrapper.find('Tag').simulate('click');
  //       expect(dispatch.callCount).toEqual(1);
  //     });
  //     ``;
  //   });
  // });
});
// describe('Input', () => {
//   let Input;
//   beforeEach(() => {
//     wrapper.setProps({ inputVisible: true });
//     Input = wrapper.find('Input');
//   });
//   it('render ', () => {
//     expect(Input.length).toEqual(1);
//   });
//
//   describe('交互行为', () => {
//     describe('聚焦时可正常编辑内容', () => {
//       it('newTagOnInput should run when change', () => {
//         Input.simulate('change', { target: { value: '23' } });
//         expect(wrapper.state('text')).toEqual('23');
//       });
//     });
//     describe('修改完毕后按 Enter 保存修改,并结束创建标签', () => {
//       it('dispatch should run once when text is empty & press Enter', () => {
//         wrapper.setState({ text: '' });
//         Input.simulate('keydown', { key: 'Enter', preventDefault: () => {} });
//         expect(dispatch.callCount).toEqual(1);
//       });
//       it('dispatch should run twice when text is not empty & press Enter', () => {
//         wrapper.setState({ text: 'test' });
//         Input.simulate('keydown', { key: 'Enter', preventDefault: () => {} });
//         expect(dispatch.callCount).toEqual(2);
//       });
//     });
//     describe('修改完毕后按Tab保存修改,并继续创建新标签', () => {
//       it('dispatch should run once when text is not empty & press Tab', () => {
//         wrapper.setState({ text: '3223' });
//         Input.simulate('keydown', { key: 'Tab', preventDefault: () => {} });
//         expect(dispatch.callCount).toEqual(1);
//       });
//       it('dispatch should not run when text is empty & press Tab', () => {
//         wrapper.setState({ text: '' });
//         Input.simulate('keydown', { key: 'Tab', preventDefault: () => {} });
//         expect(dispatch.callCount).toEqual(0);
//       });
//     });
//
//     describe('按 ESC 撤销修改并退出', () => {
//       it('cancelVOnEsc should run when press esc', () => {
//         Input.simulate('keydown', { key: 's' });
//         Input.simulate('keydown', { key: 'Escape' });
//         expect(dispatch.callCount).toEqual(1);
//       });
//     });
//
//     describe('鼠标单击空白时取消激活，并保存修改内容', () => {
//       it('newTagOnBlur should run when blur', () => {
//         wrapper.setState({ text: '3223' });
//         Input.simulate('blur');
//         expect(dispatch.callCount).toEqual(2);
//         expect(wrapper.state('text')).toEqual('');
//       });
//     });
//   });
// });
