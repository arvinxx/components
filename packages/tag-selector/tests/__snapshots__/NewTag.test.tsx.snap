// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag render 1`] = `
<div>
  <span
    class="ant-tag"
  >
    <span
      aria-label="plus"
      class="anticon anticon-plus"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="plus"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <defs>
          <style />
        </defs>
        <path
          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
        />
        <path
          d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
        />
      </svg>
    </span>
  </span>
</div>
`;
